seed/migrations/0147_import.py

```diff
@@ -0,0 +1,66 @@
+# API: "https://data.cityofnewyork.us/resource/ipu4-2q9a.json"
+# Import Permit Issuance API data into permit_issuance table
+
+import pandas as pd
+from sodapy import Socrata
+# For data abstraction - sqlalchemy
+from sqlalchemy import create_engine 
+# Driver library for database - psychopg2
+import psycopg2
+from dotenv import load_dotenv
+import os
+
+# API and Token
+load_dotenv()
+appToken = os.getenv('permitApiKey')
+client = Socrata("data.cityofnewyork.us", appToken)
+
+
+# First 2000 results, returned as JSON from API / converted to Python list of dictionaries by sodapy.
+PERMIT_ISSUANCE = "ipu4-2q9a"
+results = client.get(PERMIT_ISSUANCE, limit=2000)
+
+# Convert to pandas DataFrame
+results_df = pd.DataFrame.from_records(results)
+
+# Query fields needed for the permit_issuance table
+table_df = results_df.loc[:, ['bin__', 'job__', 'work_type', 'permit_status', 'permit_subtype']]
+
+# Specify connection parameters
+param_dic = {
+    "host"      : "localhost",
+    "database"  : "TestImport",
+    "user"      : "postgres",
+    "password"  : "postgres"
+}
+
+# Connect to the PostgreSQL database server
+conn = None
+try:
```

Comment:
I think I understand why you think you need both after reading this: https://stackoverflow.com/questions/9353822/connecting-postgresql-with-sqlalchemy

Even if sqlalchemy is using psycopg2 as the underlying library, it should be encapsulated within the library itself. Have you tried removing line 39 to 47 and see if your code actually works?

Reply:
As far as I understood from resources online, both are needed because sqlalchemy is used to abstract the data and psychopg2 is used as the driver library for the database.

Reply:
The two parts of the code just don't connect. sqlalchemy can use psycopg2 underneath for internal DB connection without exposing it to the users. I doubt you need line 39 to 47. I would suggest you actually test the code without it and observe the behavior.

Reply:
As for mapping the API data (which is converted to a pandas dataframe in the code) to the postgresql table, this resource tells me that the pandas to_sql() method maps based on column name.
https://stackoverflow.com/questions/34771256/how-does-pandas-to-sql-determine-what-dataframe-column-is-placed-into-what-datab 
Hence, I will change the column names of the df to map the columns of the postgresql table.

Reply:
Ok, sounds good about the panda thing. The API has more fields than the table. Would it automatically ignore the extra df columns if the table doesn't have those columns?

Reply:
Okay sure I will test it without lines 39 to 47.
And the API has the extra columns but the df has only the columns we need to import. We are importing the df to the table.

Reply:
Ah ok. I missed the filtering line. Thanks for clarifying